R/utilities.R

```diff
@@ -1,133 +1,158 @@
-#' @param ... Other parameters accepted by the review_hotspots() function
+#' @param genes_of_interest List of genes for hotspot review. Currently only FOXO1, MYD88, and CREBBP are supported.
+#' @param genome_build Reference genome build for the coordinates in the MAF file. The default is hg19 genome build.
+#' @param include_silent Logical parameter indicating whether to include siment mutations into coding mutations. Default is TRUE.
 #'
-#' @return
+#' @return Data frame.
 #' @export
 #'
 #' @examples
-#' coding_tabulated_df = get_coding_ssm_status(gene_symbols=c("MYC","KMT2D"))
+#' coding_tabulated_df = get_coding_ssm_status(maf_data = "maf", gene_symbols=c("MYC","KMT2D"))
 #' coding_tabulated_df = get_coding_ssm_status() #all lymphoma genes from bundled NHL gene list
+#'
 get_coding_ssm_status = function(gene_symbols,
-                                  these_samples_metadata,
-                                  from_flatfile=TRUE,
-                                  maf_path = NULL,
-                                  maf_data,
-                                  include_hotspots=TRUE,
-                                  recurrence_min = 5,
-                                  review_hotspots=TRUE,
-                                  genes_of_interest = c("FOXO1", "MYD88", "CREBBP"),
-                                  genome_build = "hg19",
-                                  include_silent=TRUE){
+                                 these_samples_metadata,
+                                 from_flatfile = TRUE,
+                                 maf_path = NULL,
+                                 maf_data,
+                                 include_hotspots = TRUE,
+                                 recurrence_min = 5,
+                                 review_hotspots = TRUE,
+                                 genes_of_interest = c("FOXO1", "MYD88", "CREBBP"),
+                                 genome_build = "hg19",
+                                 include_silent = TRUE){
+
   if(missing(gene_symbols)){
     message("defaulting to all lymphoma genes")
     gene_symbols = pull(lymphoma_genes,Gene)
   }
   if(missing(these_samples_metadata)){
     these_samples_metadata = get_gambl_metadata()
   }
-
-  coding_class = c("Frame_Shift_Del",
-                   "Frame_Shift_Ins",
-                   "In_Frame_Del",
-                   "In_Frame_Ins",
-                   "Missense_Mutation",
-                   "Nonsense_Mutation",
-                   "Nonstop_Mutation",
-                   "Silent",
-                   "Splice_Region",
-                   "Splice_Site",
-                   "Targeted_Region",
-                   "Translation_Start_Site")
   if(!include_silent){
     coding_class=coding_class[coding_class != "Silent"]
   }
-
   # call it once so the object can be reused if user wants to annotate hotspots
-  if(! missing(maf_data)){
+  if(!missing(maf_data)){
     coding_ssm = maf_data %>%
```

Comment:
Confirm whether this overwrites/clobbers the global version of this variable

Reply:
It does not.